artifactory/commands/utils/conditionaluploadutils.go

```diff
@@ -9,12 +9,16 @@ import (
 	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
 	clientutils "github.com/jfrog/jfrog-client-go/utils"
 	"github.com/jfrog/jfrog-client-go/utils/errorutils"
-	"github.com/jfrog/jfrog-client-go/utils/log"
 )
 
+type ConditionalUploadScanFuncType func(serverDetails *config.ServerDetails, fileSpec *spec.SpecFiles, threads int, scanOutputFormat xrutils.OutputFormat) error
+
+// Function to run as a condition to upload. If not overridden, the default scan function is used.
+var ConditionalUploadScanFunc ConditionalUploadScanFuncType = conditionalUploadDefaultScanFunc
+
 // ScanDeployableArtifacts scans all files founds in the given parsed deployableArtifacts results.
-// If the scan passes, the method returns two filespec ready for upload, the first one contains all the binaries
-// and the seconde all the pom.xml's.
+// If the scan passes, the method returns two file-spec ready for upload, the first one contains all the binaries
```

Comment:
method --> function
two file-spec --> two file-specs
...upload, the... --> ...upload. The...